colibri/src/icons.rs

```diff
@@ -119,6 +136,98 @@ async fn query_token_icon_and_extension(
     None
 }
 
+async fn query_uniswap_v3_position_icon(
+    chain_id: u64,
+    token_id: &str,
+    contract_address: Address,
+    inquirer: Arc<EvmNodeInquirer>,
+) -> Option<(Bytes, &'static str)> {
+    if !inquirer.connected_to_any_provider().await {
+        debug!("No connected nodes available for chain ID {} ({})", chain_id, inquirer.blockchain.as_str());
+        return None;
+    }
+
+    let token_id: U256 = match token_id.parse() {
+        Ok(id) => id,
+        Err(e) => {
+            error!("Invalid token ID '{}' for NFT position on chain ID {} ({}): {}", token_id, chain_id, inquirer.blockchain.as_str(), e);
+            return None;
+        }
+    };
```

Comment:
this is the same
```suggestion
    let token_id: U256 = token_id.parse()
    .map_err(|e| {
        error!("Invalid token ID '{}' for NFT position on chain ID {} ({}): {}", 
            token_id, chain_id, inquirer.blockchain.as_str(), e)
    })
    .ok()?;
```

and avoids the Ok(id) => id part